kernel/Dockerfile

```diff
@@ -61,8 +65,17 @@ RUN curl -fsSLO ${KERNEL_SHA256_SUMS} && \
     gpg2 --verify linux-${KERNEL_VERSION}.tar.sign linux-${KERNEL_VERSION}.tar && \
     cat linux-${KERNEL_VERSION}.tar | tar --absolute-names -x && mv /linux-${KERNEL_VERSION} /linux
 
-# Apply local patches if present
 WORKDIR /linux
+# Apply Preempt-RT kernel patches in the case of RT Linux Kernel
+RUN set -e && \
+    if [ -d /patches-${KERNEL_SERIES}${EXTRA} ]; then \
+	echo "Building ${EXTRA} kernel"; \
+	for line in `sed -e "s/#.*//g" /patches-${KERNEL_SERIES}${EXTRA}/series`; do \
```

Comment:
why is this `sed` command needed. Can't the patches just be applied like below?

Reply:
Here the 'series' file is extracted by patches-4.14.15-rt13.tar.gz. But it is originally constructed like,
###########################################################
# DELTA against a known Linus release
###########################################################

############################################################
# UPSTREAM changes queued
############################################################
rcu-Suppress-lockdep-false-positive-boost_mtx-compla.patch

############################################################
# UPSTREAM FIXES, patches pending
############################################################

############################################################
# Stuff broken upstream, patches submitted
############################################################

# soft hrtimer patches (v4)
0001-timers-Use-static-keys-for-migrate_enable-nohz_activ.patch
...
# ...
....

So we have to clean out this file. I think this should be good, otherwise we have to reconstruct our own 'series' file each time.

Reply:
ah, I hadn't realised this. This seems pretty specific to `-rt` could we make this section so that we look for `series` and if present do what you do and otherwise, just apply `*.patch`? That way we can re-use the mechanism for patching